cpp/src/arrow/array/data.h

```diff
@@ -434,6 +435,23 @@ struct ARROW_EXPORT ArraySpan {
     return GetValues<T>(i, this->offset);
   }
 
+  // Access a buffer's data as a span
+  template <typename T>
+  util::span<const T> GetSpan(int i, int64_t length) const {
+    int buffer_length = static_cast<size_t>(buffers[i].size) / sizeof(T);
+    assert(length > 0 && length + offset <= buffer_length);
```

Comment:
`0` is a valid length for some buffers, did you mean to check `i > 0`? (because `0` is a bitmap buffer and can't be accessed as a `span`).